cosmos-cmd-tlm-api/app/controllers/storage_controller.rb

```diff
@@ -29,6 +29,8 @@ def get_download_presigned_request
       render(:json => { :status => 'error', :message => e.message }, :status => 403) and return
     end
 
+    @rubys3_client = Aws::S3::Client.new
+    @rubys3_client.head_object(bucket: params[:bucket], key: params[:object_id])
```

Comment:
Should the get_presigned_request method just do the head_object? Then you can remove this and the client initialization. It's already doing head_bucket.

Reply:
Needs to be here because put doesn't head_object